File: .codex-presentation-work/gemini-only-20260731/bin/unzip.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int main(int argc, char **argv) {
    char command[32768];
    if (argc == 3 && strcmp(argv[1], "-Z1") == 0) {
        snprintf(command, sizeof(command), "tar -tf \"%s\"", argv[2]);
        return system(command);
    }
    if (argc == 4 && strcmp(argv[1], "-p") == 0) {
        snprintf(command, sizeof(command), "tar -xOf \"%s\" \"%s\"", argv[2], argv[3]);
        return system(command);
    }
    fprintf(stderr, "Unsupported unzip arguments\n");
    return 2;
}

File: .codex-presentation-work/gemini-only-20260731/bin/unzip.mjs
import { spawnSync } from "node:child_process";

const args = process.argv.slice(2);
let tarArgs;
if (args[0] === "-Z1" && args.length === 2) {
  tarArgs = ["-tf", args[1]];
} else if (args[0] === "-p" && args.length === 3) {
  tarArgs = ["-xOf", args[1], args[2]];
} else {
  process.stderr.write(`Unsupported unzip arguments: ${args.join(" ")}\n`);
  process.exit(2);
}

const result = spawnSync("tar", tarArgs, { encoding: null, maxBuffer: 100 * 1024 * 1024 });
if (result.stdout) process.stdout.write(result.stdout);
if (result.stderr) process.stderr.write(result.stderr);
process.exit(result.status ?? 1);
